idb/postgres/postgres.go

```diff
@@ -1222,6 +1232,11 @@ ON CONFLICT (addr, assetid) DO UPDATE SET amount = account_asset.amount + EXCLUD
 			return
 		}
 	}
+	if istate.AccountRound >= int64(round) {
```

Comment:
I think there's a race condition here. If both transactions run the select before the upsert then they could both still be committed.

What do you think about
1) Removing the metastate query and setting the account round directly with `WHERE round < $1`.
2) Returning an error if the updated rows equal zero (meaning another transaction beat us to the punch)